### PR TITLE
Small tweaks to journald logging

### PR DESCRIPTION
**Pull request by haircommander**
Two changes to better match docker functionality:
	Change MESSAGE_ID to CONTAINER_ID_FULL, as well as add CONTAINER_ID (truncated at 12 characters)
	Change stdout output to an info instead of a notice log level
Also small change with where the log buffer is null terminated
	Now, the callee doesn't depend on an unchecked invariant that num_read < strlen(buf)

Signed-off-by: Peter Hunt <pehunt@redhat.com>

**- What I did**

**- How I did it**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

**Comment by haircommander** [comment]
@giuseppe @mrunalp @rhatdan PTAL

**Comment by umohnani8** [comment]
changes LGTM

File: conmon/ctr_logging.c [hunk 1]
```diff
@@ -74,10 +76,12 @@ void configure_log_drivers(gchar **log_drivers, int64_t log_size_max_, char *cuu
 		nexit("Include journald in compilation path to log to systemd journal");
 #endif
 		cuuid = cuuid_;
+		if (strlen(cuuid) <= TRUNCIDLEN)
+			nexit("Container ID is too short");
+		strncpy(short_cuuid, cuuid_, TRUNCIDLEN);
```
Comment (giuseppe) [inline conmon/ctr_logging.c]: `strncpy` won't add a '\0' if it is not in the first TRUNCIDLEN bytes of cuuid_.  Please add it explicitly with `short_cuuid[TRUNCIDLEN] = '\0';`

Comment (haircommander) [reply conmon/ctr_logging.c]: fixed!

**Comment by haircommander** [comment]
/hold
(I have one more thing I'm trying to implement)

**Comment by haircommander** [comment]
Note: I'd love to have a test for CONTAINER_PARTIAL_LINE but the only way I have been able to cause it is by running podman with a top container and sending it sigkill. Any advice would be nice :)

File: conmon/ctr_logging.c [hunk 2]
```diff
@@ -121,16 +129,38 @@ bool write_to_logs(stdpipe_t pipe, char *buf, ssize_t num_read)
...
 	if (pipe == STDERR_PIPE)
 		message_priority = LOG_ERR;
-	sd_journal_send("MESSAGE=%s", buf, "PRIORITY=%i", message_priority, "MESSAGE_ID=%s", cuuid, NULL);
+
+    ptrdiff_t line_len = 0;
+    while (buflen > 0) {
+        bool partial = get_line_len(&line_len, buf, buflen);
+        /* sd_journal_* doesn't have an option to specify the number of bytes to write, and instead writes the
+         * entire string. Copying every line doesn't make very much sense, so instead we do this tmp_line_end
+         * hack to emulate seperate strings.
```
Comment (giuseppe) [inline conmon/ctr_logging.c]: typo in `seperate`

**Comment by giuseppe** [comment]
> Note: I'd love to have a test for CONTAINER_PARTIAL_LINE but the only way I have been able to cause it is by running podman with a top container and sending it sigkill. Any advice would be nice :)

I was testing partial lines with something similar to podman run fedora sh -c 'printf hello; sleep 1'

File: conmon/conmon.c [hunk 3]
```diff
@@ -975,7 +980,6 @@ int main(int argc, char *argv[])
 		exit(0);
 	}
 
-	// why not nexit?
 	if (opt_cid == NULL) {
 		fprintf(stderr, "Container ID not provided. Use --cid\n");
```
Comment (umohnani8) [inline conmon/conmon.c]: Not related to this PR, but you could probably use `nexit` here instead to match the others below it.

Comment (haircommander) [reply conmon/conmon.c]: It turns out nexit uses opt_cid to print ctrID with the error; if it isn't provided it would segfault

Comment (umohnani8) [reply conmon/conmon.c]: oh okay, got it! thanks for the info :)

**Comment by haircommander** [comment]
Huh I have the label And happy tests... Can someone try to trigger it again?

**Comment by haircommander** [comment]
I seek approval 
@umohnani8 @mrunalp @giuseppe

**Comment by mrunalp** [comment]
LGTM

File: conmon/ctr_logging.c [hunk 4]
```diff
@@ -156,18 +187,9 @@ static int write_k8s_log(stdpipe_t pipe, const char *buf, ssize_t buflen)
...
-		bool partial = FALSE;
-
-		/* Find the end of the line, or alternatively the end of the buffer. */
-		line_end = memchr(buf, '\n', buflen);
-		if (line_end == NULL) {
-			line_end = &buf[buflen - 1];
-			partial = TRUE;
-		}
-		line_len = line_end - buf + 1;
+		bool partial = get_line_len(&line_len, buf, buflen);
```
Comment (TomSweeneyRedHat) [inline conmon/ctr_logging.c]: nit of a nit, but I don't think you need to declare this as bool here.

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by haircommander** [comment]
Can someone throw me a /approve? :)

File: conmon/ctr_logging.c (hunk 4 above)
Comment (haircommander) [reply conmon/ctr_logging.c]: Why not? It's a previously undeclared variable :)

**Comment by TomSweeneyRedHat** [comment]
Not sure I've the juice to /approve, but we'll see.

**Pull request merged**